crowbar_framework/app/models/batch/export.rb

```diff
@@ -0,0 +1,177 @@
+#
+# Copyright 2011-2013, Dell
+# Copyright 2013-2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Batch
+  class Export < Base
+    def path
+      @path ||= Rails.root.join(
+        "storage",
+        "batch",
+        "export",
+        filename
+      )
+
+      unless @path.dirname.directory?
+        @path.dirname.mkpath
+      end
+
+      @path
+    end
+
+    def filename
+      @filename ||= begin
+        time = Time.now.utc.strftime(
+          "%Y%m%d-%H%M%S%z"
+        )
+
+        "batch-#{time}.yml"
+      end
+    end
+
+    protected
+
+    def persist!
+      path.open(File::CREAT | File::TRUNC | File::RDWR) do |file|
+        file.write content.to_yaml
+      end
+
+      true
+    end
+
+    def content
+      struct = [].tap do |result|
+        barclamps.each do |barclamp|
+          next unless process_barclamp?(
+            barclamp
+          )
+
+          value = process_barclamp!(
+            barclamp
+          )
+
+          result.concat(
+            value
+          ) if value.present?
+        end
+      end
+
+      {
+        "proposals" => struct
+      }
+    end
+
+    def process_barclamp?(barclamp)
+      if excluder = expanded_includes.present? \
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `excluder`. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)